src/Symfony/Component/Validator/Command/DebugCommand.php

```diff
@@ -0,0 +1,192 @@
+
+        foreach ($this->getConstrainedPropertiesData($class) as $propertyName => $constraintsData) {
+            foreach ($constraintsData as $data) {
+                $rows[] = [
+                    $propertyName,
+                    $data['class'],
+                    implode(', ', $data['groups']),
+                    $dump($data['options']),
+                ];
+            }
+        }
+
+        if (empty($rows)) {
+            if (false === $input->getOption('show-all')) {
+                return;
+            }
+
+            $io->title($title);
+            $io->text('No validators were found for this class.');
+
+            return;
+        }
+
+        $io->title($title);
+
+        $table = new Table($output);
+        $table->setHeaders(['Property', 'Name', 'Groups', 'Options']);
+        $table->setRows($rows);
+        $table->setColumnMaxWidth(3, 80);
+        $table->render();
+    }
+
+    private function getConstrainedPropertiesData(string $class): array
+    {
+        $data = [];
+
+        /** @var ClassMetadataInterface $classMetadata */
+        $classMetadata = $this->validator->getMetadataFor($class);
+
+        foreach ($classMetadata->getConstrainedProperties() as $constrainedProperty) {
+            $data[$constrainedProperty] = $this->getPropertyData($classMetadata, $constrainedProperty);
+        }
+
+        return $data;
+    }
+
+    private function getPropertyData(ClassMetadataInterface $classMetadata, string $constrainedProperty): array
+    {
+        $data = [];
+
+        $propertyMetadata = $classMetadata->getPropertyMetadata($constrainedProperty);
+        foreach ($propertyMetadata as $metadata) {
+            foreach ($metadata->getConstraints() as $constraint) {
+                $data[] = [
+                    'class' => \get_class($constraint),
+                    'groups' => $constraint->groups,
+                    'options' => $this->getConstraintOptions($constraint),
+                ];
+            }
+        }
+
+        return $data;
+    }
+
+    private function getConstraintOptions(Constraint $constraint): array
+    {
+        $options = [];
+
+        $constraintClass = \get_class($constraint);
+
+        foreach (array_keys(get_class_vars($constraintClass)) as $propertyName) {
+            try {
+                $options[$propertyName] = $constraint->$propertyName;
+            } catch (\Exception $exception) {
+            }
+        }
+
+        return $options;
+    }
+
+    private function getResourcesByPath(string $path): array
+    {
+        $finder = new Finder();
+        $finder->files()->in($path)->name('*.php')->sortByName(true);
+        $classes = [];
+
+        foreach ($finder as $file) {
+            $fileContent = file_get_contents($file->getRealPath());
+
+            preg_match('/namespace (.+);/', $fileContent, $matches);
+
+            $namespace = $matches[1] ?? null;
+
+            if (false == preg_match('/class (.+)/', $fileContent, $matches)) {
```

Comment:
What about `class +([^{ ]+)` ? So you don't need the explode below, just use `$matches[1]`

Reply:
@ogizanagi You're right, I made that change.